modules/subschema/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/subschema/package.scala

```diff
@@ -3,26 +3,56 @@ package com.snowplowanalytics.iglu.schemaddl.jsonschema.subschema
 import com.snowplowanalytics.iglu.schemaddl.jsonschema.Schema
 import com.snowplowanalytics.iglu.schemaddl.jsonschema.properties.CommonProperties._
 import com.snowplowanalytics.iglu.schemaddl.jsonschema.properties.CommonProperties.Type._
-
 import dregex.Regex
 import io.circe.Json
 
+import scala.annotation.tailrec
+
 package object subschema {
 
   def isSubSchema(s1: Schema, s2: Schema): Compatibility =
     isSubType(simplify(canonicalize(s1)), simplify(canonicalize(s2)))
 
+  @tailrec
   def canonicalize(s: Schema): Schema =
     s match {
-      case s if s.`type`.contains(Null)    => s
-      case s if s.`type`.contains(Boolean) =>
-        if (s.`enum`.isDefined) s else s.copy(`enum` = Some(Enum(List(Json.True, Json.False))))
-      case s if s.`type`.contains(Integer) => s.copy(`type` = Some(Number))
-      case s if s.`type`.contains(Number)  => s
-      case s if s.`type`.contains(String)  => s
+      case s if s.`type`.contains(Null)                => s.copy(`enum` = None)
```

Comment:
When encountering a schema with type null, we can discard the enum information, as the only possible value that matches the schema is the null value itself